config/crd/bases/keda.sh_clustertriggerauthentications.yaml

```diff
@@ -91,11 +91,23 @@ spec:
                         required:
                         - valueFrom
                         type: object
+                      podIdentity:
+                        description: AuthPodIdentity allows users to select the platform
+                          native identity mechanism
+                        properties:
+                          identityId:
+                            type: string
+                          provider:
+                            description: PodIdentityProvider contains the list of
+                              providers
+                            type: string
+                        required:
+                        - provider
+                        type: object
                       tenantId:
                         type: string
                     required:
                     - clientId
-                    - clientSecret
                     - tenantId
```

Comment:
Why are the `clientId`, and `tenantId` required? With the way it was structured earlier, `clientSecret` was required because credentials could only be used with scerets. In case of pod identity credentials had to be omitted. 

But now that users can specify separate pod identity for key vault, I think no property within the credentials needs to be "required".

Related to this, https://github.com/kedacore/keda/blob/5ca11196f54a26efa4cdda38fc5ec283f167286c/config/crd/bases/keda.sh_clustertriggerauthentications.yaml#L129-L131

I think `credentials` now becomes a required property, and needs to be added here, if I am not wrong.

Reply:
that's true, with pod identity we don't need the client_id or tenant_id. good catch!
I'm thinking on moving the `podIdentity` section at same level of `credentials`, keeping `credentials` section as it was and making the new `podIdentity` section optional too

Reply:
That makes quite a bit of sense actually, as you can argue `podIdentity` isn't exactly a credential for it to be inside `credentials`.